### ISSUE TITLE
does it handles reconnections?

### ISSUE THREAD

**Issue by dcu** [daneharrigan/hipchat#6]
If not, how can I do it?

**Comment by cjhubert** [comment]
I'm also curious about reconnections. I noticed my bot would lose connections every >100 seconds and looked through the code. Noticed I should be doing something like `go client.KeepAlive()` after making the client to make sure it doesn't lose connection. However, it still sometimes seems to disconnect while my other hipchat bots on the same machine (in other languages) still work.

**Comment by daneharrigan** [comment]
@dcu hey sorry for the delay. you're using `KeepAlive()` and it still disconnects you? are you connecting with the same JID and resource elsewhere? hipchat will boot accounts if that happens

**Comment by daneharrigan** [comment]
i created a sample, but i was able to reproduce the problem:

```go
package main

import (
	"github.com/daneharrigan/hipchat"
	"fmt"
	"time"
)

func main() {
	user := "redacted"
	pass := "redacted"
	resource := "bot"
	roomJid := "redacted"
	fullName := "Dane Harrigan"

	client, err := hipchat.NewClient(user, pass, resource)
	if err != nil {
		fmt.Printf("client error: %s\n", err)
		return
	}

	client.Status("chat")
	client.Join(roomJid, fullName)
	go client.KeepAlive()

	// output messages you receive
	go func() {
		for m := range client.Messages() {
			fmt.Printf("%s: %s\n", m.From, m.Body)
		}
	}()

	// send "ping" messages to show you're still alive
	for {
		client.Say(roomJid, fullName, "ping")
		time.Sleep(2 * time.Minute) // wait 120sec between pings because the problem occurs >100sec 
	}
}
```

i added some comments in the code to indicate what i'm doing and why. i slept for 120sec between pings to exceed the 100sec issue, but i haven't encountered any disconnects. can  you send code to reproduce the problem?

**Comment by cjhubert** [comment]
The exact problem I was experiencing has been given more detail in #8 by @fedyakin. Making the changes spoken about (changing `Rooms()` and `Users()` to return the channel and exposing the `RequestRooms() and RequestUsers()`) fixed the issue we were seeing where eventually the bot would get an `iq` it wasn't expecting and just hang.

**Comment by daneharrigan** [comment]
fixed in 9b8743b68827e88faa248bc432197c570871117d
